@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -828,8 +823,6 @@ export function placePathLabel(
         checkGlyphCollision = candidateBoxes.length > 0;
     }
 
-    const checkGlyphVisible = !screenCollisions.isFullyVisible(tmpCollisionBox);
-
```

Comment:
I guess you removed this because you're only passing visible path points from isPathLabelTooSmall. That's wrong, check my comment there.
As long as we pass here invisible path points, we need these checks. Another option is to clamp the invisible points to the screen borders and the I guess we can get rid of this.

But frankly i wouldn't do any further changes as part of this PR, is too difficult to follow all changes already.